modules/cdc-ext/src/test/java/org/apache/ignite/cdc/CacheConflictOperationsTest.java

```diff
@@ -0,0 +1,267 @@
+
+        for (CacheAtomicityMode mode : EnumSet.of(ATOMIC, TRANSACTIONAL))
+            for (byte otherClusterId : new byte[] {FIRST_CLUSTER_ID, THIRD_CLUSTER_ID})
+                params.add(new Object[] {mode, otherClusterId});
+
+        return params;
+    }
+
+    /** */
+    private static IgniteCache<String, Data> cache;
+
+    /** */
+    private static IgniteInternalCache<BinaryObject, BinaryObject> cachex;
+
+    /** */
+    private static IgniteEx client;
+
+    /** Cluster have a greater priority that {@link #SECOND_CLUSTER_ID}. */
+    private static final byte FIRST_CLUSTER_ID = 1;
+
+    /** */
+    private static final byte SECOND_CLUSTER_ID = 2;
+
+    /** Cluster have a lower priority that {@link #SECOND_CLUSTER_ID}. */
+    private static final byte THIRD_CLUSTER_ID = 3;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        CacheVersionConflictResolverPluginProvider<?> pluginCfg = new CacheVersionConflictResolverPluginProvider<>();
+
+        pluginCfg.setClusterId(SECOND_CLUSTER_ID);
+        pluginCfg.setCaches(new HashSet<>(Collections.singleton(DEFAULT_CACHE_NAME)));
+        pluginCfg.setConflictResolveField(conflictResolveField());
+
+        return super.getConfiguration(igniteInstanceName)
+            .setPluginProviders(pluginCfg);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        startGrid(1);
+        client = startClientGrid(2);
+
+        cache = client.createCache(new CacheConfiguration<String, Data>(DEFAULT_CACHE_NAME).setAtomicityMode(cacheMode));
+        cachex = client.cachex(DEFAULT_CACHE_NAME);
+    }
+
+    /** Tests that regular cache operations works with the conflict resolver when there is no update conflicts. */
+    @Test
+    public void testSimpleUpdates() {
+        String key = "UpdatesWithoutConflict";
+
+        put(key);
+
+        put(key);
+
+        remove(key);
+    }
+
+    /**
+     * Tests that {@code IgniteInternalCache#*AllConflict} cache operations works with the conflict resolver
+     * when there is no update conflicts.
+     */
+    @Test
+    public void testUpdatesFromOtherClusterWithoutConflict() throws Exception {
+        String key = "UpdateFromOtherClusterWithoutConflict";
+
+        putx(key(key, otherClusterId), otherClusterId, 1, true);
+
+        putx(key(key, otherClusterId), otherClusterId, 2, true);
+
+        removex(key(key, otherClusterId), otherClusterId, 3, true);
+    }
+
+
+    /**
+     * Tests that {@code IgniteInternalCache#*AllConflict} cache operations works with the conflict resolver
+     * when there is no update conflicts.
+     */
+    @Test
+    public void testUpdatesReorderFromOtherCluster() throws Exception {
+        String key = "UpdateClusterUpdateReorder";
+
+        putx(key(key, otherClusterId), otherClusterId, 2, true);
+
+        // Update with the equal or lower order should fail.
```

Comment:
How about a test where the order the same, but the field to compare is greater and vice versa?

Also, I'm not sure that order (which is global, not per entry, should win), we expect version comparison, not orders.